libsql/src/sync.rs

```diff
@@ -75,4 +99,46 @@ impl SyncContext {
     pub(crate) fn durable_frame_num(&self) -> u32 {
         self.durable_frame_num
     }
+
+    async fn write_metadata(&mut self) -> Result<()> {
+        let path = format!("{}-info", self.db_path);
+
+        let contents = serde_json::to_vec(&MetadataJson {
+            version: METADATA_VERSION,
+            durable_frame_num: self.durable_frame_num,
+        })
+        .unwrap();
+
+        tokio::fs::write(path, contents).await.unwrap();
+
+        Ok(())
+    }
+
+    async fn read_metadata(&mut self) -> Result<()> {
+        let path = format!("{}-info", self.db_path);
+
+        if !std::fs::exists(&path).unwrap() {
+            tracing::debug!("no metadata info file found");
+            return Ok(());
+        }
+
+        let contents = tokio::fs::read(&path).await.unwrap();
+
+        let metadata = serde_json::from_slice::<MetadataJson>(&contents[..]).unwrap();
+
+        assert_eq!(
+            metadata.version, METADATA_VERSION,
+            "Reading metadata from a different version than expected"
+        );
+
+        self.durable_frame_num = metadata.durable_frame_num;
+
+        Ok(())
+    }
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+struct MetadataJson {
+    version: u32,
+    durable_frame_num: u32,
```

Comment:
Should we make it u64 or even u128 to make it more future proof?

Reply:
These were selected by @penberg so he should comment on making those changes.

Reply:
We should add checkpoint generation there

Reply:
I am adding the tracking code for this but atm we only have generation set to 1.

Reply:
https://github.com/tursodatabase/libsql/pull/1830/commits/3575914dbaa856eaaa59f9fe070c48eca206e250